legion_core/src/storage.rs

```diff
@@ -913,6 +913,28 @@ impl ArchetypeData {
     ) -> &mut Chunkset {
         self.chunksets_mut().get_unchecked_mut(index)
     }
+
+    pub(crate) fn iter_data_slice<'a, T: Component>(
+        &'a self,
+    ) -> impl Iterator<Item = RefMap<&[T]>> + 'a {
+        self.chunk_sets.iter().flat_map(move |set| {
+            set.chunks.iter().map(move |chunk| {
+                let c = chunk.components(ComponentTypeId::of::<T>()).unwrap();
+                unsafe { c.data_slice::<T>() }
+            })
+        })
+    }
+
+    pub(crate) fn iter_data_slice_mut<'a, T: Component>(
```

Comment:
The runtime borrow checking is disabled in release mode and so cant be relied upon for safety guarentees. This needs to either be `unsafe` or require `&mut self`.

Reply:
I marked the function as unsafe (and renamed to iter_data_slice_unchecked_mut)